src/organization/screens/organization-profile.screen.js

```diff
@@ -199,6 +184,38 @@ class OrganizationProfile extends Component {
   }
 }
 
-export const OrganizationProfileScreen = connect(selectors, actions)(
-  OrganizationProfile
-);
+const mapStateToProps = (state, ownProps) => {
+  const {
+    auth: { user, locale },
+    pagination: { ORGS_GET_MEMBERS },
+    entities: { orgs, users },
+  } = state;
+
+  const orgId = ownProps.navigation.state.params.organization.login;
+  const org = orgs[orgId] || ownProps.navigation.state.params.organization;
```

Comment:
**Something important to have in mind:**

This screen is accessed from a User profile, or from the Search results. 

In both cases, `ownProps.navigation.state.params.organization`  contains an incomplete Organization object (left), which is not enough to render the screen, as we need the number of repositories, etc, which is only present after calling the organization endpoint to get the complete object (right):

| initial object | after fetching information |
| ------------ | -------------------------- |
| <img width="791" alt="screen shot 2018-03-27 at 11 49 31 pm" src="https://user-images.githubusercontent.com/304450/37999165-8984b688-3219-11e8-898a-e80c86d5f058.png"> | <img width="798" alt="screen shot 2018-03-27 at 11 51 22 pm" src="https://user-images.githubusercontent.com/304450/37999270-fc17d86a-3219-11e8-9d8e-92e9f658c3e0.png"> |

This is why, _with or without this PR_, if you navigate to a user profile, turn off internet, and then click on one of his organizations, you will see a broken incomplete UI.  Usually the GitHub API call is so quick that it goes unnoticed.

---

Normally, `org` would only be read from `state.orgs`, which would already contain the incomplete organization object. Since the User profile & Search screens doesn't use the new API yet, hence, doesn't populate `state.orgs` with the incomplete object, I'm retrieving it from the navigation state.

Reading from navigation will be removed in a future PR, once the new Redux API is used everywhere. Only the organization **id** will be passed in the navigation object, and the Redux API will go get the object for us.

---

If we use this same approach for all our screens (receiving only IDs, no objects, and letting the redux API fetch things), GitPoint will be able to handle all Github.com urls with deep linking 🎉

Reply:
I wouldn't have done that differently as a first approach, but working with this new Redux API & having in mind deep linking, it's now obvious that every screen should be "standalone" and be able to fetch it's data with simple ids.

A pull screen for example would require a repository identifier & the pull number, just like this Pull Request url on GitHub web:
    `https://github.com/**gitpoint/git-point**/pull/**752**`

nahmean?